crates/router/src/connector/billwerk/transformers.rs

```diff
@@ -136,28 +256,68 @@ impl<F, T>
             types::PaymentsResponseData,
         >,
     ) -> Result<Self, Self::Error> {
+        let error_response = if item.response.error.is_some() || item.response.error_state.is_some()
```

Comment:
will error & error_state will be populated in case of payment failures too? or only incase of bad request?

Reply:
In case of failures too